src/dxgi/dxgi_output.cpp

```diff
@@ -93,21 +93,42 @@ namespace dxvk {
     std::vector<DXGI_MODE_DESC> modes(modesCount);
     GetDisplayModeList(pModeToMatch->Format, 0, &modesCount, modes.data());
 
+    //filter out modes with different scaling if it was set
+    if (modeToMatch.Scaling != DXGI_MODE_SCALING_UNSPECIFIED) {
+      if (modeToMatch.Scaling != DXGI_MODE_SCALING_CENTERED) {
+        Logger::warn("DxgiOutput::FindClosestMatchingMode: DXVK currently supports only DXGI_MODE_SCALING_CENTERED. Fall back to it.");
+        modeToMatch.Scaling = DXGI_MODE_SCALING_CENTERED;
+      }
+
+      for (auto it = modes.begin(); it != modes.end();) {
+         if (it->Scaling != modeToMatch.Scaling)
+           it = modes.erase(it);
+         else
+           ++it;
+      }
+    }
+
     //filter out modes with different scanline ordering if it was set
     if (modeToMatch.ScanlineOrdering != DXGI_MODE_SCANLINE_ORDER_UNSPECIFIED) {
+      if (modeToMatch.ScanlineOrdering != DXGI_MODE_SCANLINE_ORDER_PROGRESSIVE) {
+        Logger::warn("DxgiOutput::FindClosestMatchingMode: DXVK currently supports only DXGI_MODE_SCANLINE_ORDER_PROGRESSIVE. Fall back to it.");
+        modeToMatch.ScanlineOrdering = DXGI_MODE_SCANLINE_ORDER_PROGRESSIVE;
+      }
+
       for (auto it = modes.begin(); it != modes.end();) {
         if (it->ScanlineOrdering != modeToMatch.ScanlineOrdering)
```

Comment:
Same thing here, I don't think we need this (or the clunky log message) until DXVK actually reports the scaling modes and scanline modes properly.

Reply:
same, removed